src/main/java/org/folio/circulation/support/ItemRepository.java

```diff
@@ -89,7 +89,7 @@ private ItemRepository(
   }
 
   public CompletableFuture<Result<Item>> fetchFor(ItemRelatedRecord record) {
-    return fetchById(record.getItemId());
+    return fetchById(record.getItemId()).thenComposeAsync(this::fetchLocation);
```

Comment:
Can you help me understand why this has been added here?

Have all of the endpoints that use this method been checked that they have sufficient permissions to fetch locations?

Reply:
This populates `location` in items. This is based on a premise that every item _should_ have a location.
In the end it allows to add new fields to json response in /requets/{id} endpoint.

Reply:
> This is based on a premise that every item should have a location.

Where did that premise come from? 

There may be contexts where location is not needed, and the overhead of the 4 extra requests is undesirable.

Have all of the endpoints that use this method been checked for having the correct permissions to fetch locations, and the hierarchy of records?

Reply:
I have removed it and I'm now checking the required permissions.

Reply:
Thank you